docs/src/main/docs/guides/91_mp-tutorial.adoc

```diff
@@ -0,0 +1,1258 @@
+ a `Set` of `Class` objects, one for each resource that the application 
+ will serve.  Currently, the application only has one resource, the
+ `GreetResource`.
+
+TIP:  You can learn more about scopes and contexts; and how they are used from the https://javaee.github.io/javaee-spec/javadocs/javax/enterprise/context/package-summary.html[Java EE
+ Specification].
+
+The `GreetResource` is defined in the `GreetResource.java` class as shown
+below: 
+
+[source,java]
+.Initial resource class
+----
+package io.helidon.examples;
+
+import javax.enterprise.context.RequestScoped;
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.Collections;
+
+@Path("/greet") <1>
+@RequestScoped <2>
+public class GreetResource {
+
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public JsonObject getDefaultMessage() { <3>
+        return JSON.createObjectBuilder()
+                .add("message", "Hello World")
+                .build(); <4>
+    }
+
+}
+----
+
+<1> This class is annotated with `Path` which sets the path for this resource
+ as `/greet`.  
+<2> The `RequestScoped` annotation defines that this bean is
+ request scoped.  The request scope is active only for the duration of
+ one web service invocation and it is destroyed at the end of that
+ invocation.
+<3> A `public JsonObject getDefaultMessage()` method is defined
+ which is annotated with `GET`, meaning it will accept the HTTP GET method.
+ It is also annotated with `Produces(MediaType.APPLICATION_JSON)` which
+ declares that this method will return JSON data. 
+<4> The method body creates
+ a JSON object containing a single object named "message" with the content
+ "Hello World".  This method will be expanded and improved
+ later in the tutorial.
+
+A main class is also required to start up the server and run the
+ application.  Here is the initial main class:
+
+[source,java]
+.Initial Main class
+----
+package io.helidon.examples;
+
+import io.helidon.microprofile.server.Server;
+import java.io.IOException;
+
+public final class Main {
+
+    private Main() { } <1>
+
+    public static void main(final String[] args) throws IOException {
+        Server server = startServer();
+        System.out.println("http://localhost:" + server.port() + "/greet");
+    }
+
+    static Server startServer() {
+        return Server.create().start(); <2>
+    }
+    
+}
+----
+
+In this class, a `main` method is defined which starts the Helidon MP 
+ server and prints out a message with the listen address. 
+
+<1> Notice that
+ this class has an empty no-args constructor to make sure this class
+ cannot be instantiated.
+<2> The Helidon reactive web server is started with the default configuration.
```

Comment:
We are starting the Microprofile `Server`. It does start the reactive web server under the hood, though that is probably not to be mentioned here (not to confuse the reader).